docs/Building-Transmission.md

```diff
@@ -120,6 +120,90 @@ $ sudo make install
 ```
 
 ## On Windows ##
+
+## Building transmission-daemon
+You need the following installed:
+
+* Visual Studio 2017 (the Community Edition is sufficient - just make sure its C[++] compiler, MSVC, is installed)
+* [ActivePerl](https://www.activestate.com/products/activeperl/) or [StrawberryPerl](https://strawberryperl.com)
+* [CMake](https://cmake.org/download/) (choose to add CMake to your path)
+* Possibly [Git for Windows](https://git-scm.com/download/win) to have tools like `patch` present
+* [jom](https://wiki.qt.io/Jom) (very recommended, esp. for OpenSSL compilation which can take ages on a single core. If you want to use `nmake`, use "NMake Makefiles" as the CMake generator & don't add `/FS` to CFLAGS)
+
+### Set up the environment
+
+Open a Command Prompt and run the following:
+
+```
+call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsall.bat" amd64
+set "TPDIR=C:\3rd-party-msvc64"
+set "PATH=%TPDIR%\bin;%PATH%"
+set "TPCFLAGS=/nologo /MP /GS /GL /Gy /Oi /Oy /O2 /DWIN32 /D_WINDOWS"
+set "TPLDFLAGS=/NOLOGO /DYNAMICBASE /NXCOMPAT /LTCG /INCREMENTAL:NO /OPT:REF /OPT:ICF /machine:x64"
+set "CFLAGS=%TPCFLAGS%"
+set "LDFLAGS=%TPLDFLAGS%"
+
+mkdir "%TPDIR%" & cd "%TPDIR%"
+```
+
+### Build zlib
+
+Run the following to build and install zlib into %TPDIR%:
+
+```
+mkdir build & cd build
+cmake .. -G "NMake Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX="%TPDIR%" -DCMAKE_SHARED_LINKER_FLAGS="%LDFLAGS%" -DCMAKE_C_FLAGS="%CFLAGS%" -DBUILD_SHARED_LIBS=ON -DSKIP_INSTALL_FILES=ON -DAMD64=ON
```

Comment:
Please only advise `-DCMAKE_BUILD_TYPE=RelWithDebInfo` by default, as we do not want people reporting issues without the debug symbols.